src/anm/analytics.js

```diff
@@ -0,0 +1,79 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics/player');
+
+    self.enabled = animatronUrl != null && animatronUrl.indexOf('animatron-test') >= 0;
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+            } else {
+                sendViaGif(data);
+            }
+        }
+        if (!supportSendBeacon) {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (data) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = animatronUrl + '?data=' + encodeURIComponent(data);
+    };
+    if (self.enabled) {
+        if (!supportSendBeacon) {
+            engine.onDocReady(event);
```

Comment:
Sorry for being so mad. It's just so much discussion and words for such small task. Just call me next time ;)